app/views/dashboards/show.html.erb

```diff
@@ -65,7 +65,32 @@
         <button class="btn btn-default" ng-click="enableFullscreen()" title="Fullscreen mode">
           <span class="glyphicon glyphicon-fullscreen"></span>
         </button>
+        <% unless @dashboard.static %>
+          <button class="btn btn-default" ng-click="generateStaticDashboard()"
+            tooltip-append-to-body="true" tooltip="Create dashboard permalink">
+            <span ng-show="!generatingPermalink" class="glyphicon glyphicon-link"></span>
+            <span ng-show="generatingPermalink" class="glyphicon glyphicon-refresh spin"></span>
+          </button>
+        <% end %>
       </div>
+      <% unless @dashboard.static %>
+        <div class="dashboard_settings panel panel-default" ng-show="showPermalink">
+          <div class="panel-heading">Dashboard Permalink</div>
+          <div class="panel-body">
+            <div class="input-group">
+              <input type="text" class="form-control" ng-model="permalink">
+              <span class="input-group-btn">
+                <a class="btn btn-default" href="{{permalink}}"
+                  tooltip-append-to-body="true"
+                  tooltip="Open dashboard permalink in new tab" target="_blank"
+                  ng-click="showPermalink = false">
+                  <span class="glyphicon glyphicon-new-window"></span>
+                </a>
+              </span>
+            </div>
+          </div>
+        </div>
+      <% end %>
```

Comment:
can't comment on the right line, but: shouldn't the "Save" button also disappear for permlink dashboards? At least pressing it doesn't work, and I'd recommend hiding it.